simplify desktop access getting started guide (#9100)
gravitational/teleport: docs/pages/desktop-access/getting-started.mdx

- A Linux server to run the Teleport Desktop Access service on
  - You can reuse an existing server running any other Teleport instance

## Step 1/6. Create a restrictive service account `User`.

Teleport requires a service account to connect to your Active Directory domain. We recommend creating a dedicated service account with restrictive permissions
for maximal security.

To create the service account, open a PowerShell prompt and copy-paste in the commands below. Be sure to note down the password you choose (you will be prompted to
enter it in the PowerShell), as you will need to provide it to the Teleport configuration yaml later.

```powershell
$Name="Teleport Service Account"
$SamAccountName="svc-teleport"

> New-ADUser `

Reviewer: Are these '`' characters intentional? Is that how you wrap lines in powershell?

Author: [Indeed](https://stackoverflow.com/a/2608186/6277051)

Reviewer: Wild 🤯

  -Name $Name `
  -SamAccountName $SamAccountName `
  -AccountPassword (Read-Host "Set Password (and note it down)" -AsSecureString) `
  -Enabled $true
```

> Next, in the same PowerShell prompt, enter the following commands to limit your service account's TODO permissions to the bare minimum necessary for

Reviewer: ```suggestion
Next, in the same PowerShell prompt, enter the following commands to limit your service account's permissions
```

Teleport to operate.

```powershell
# Save your domain's distinguished name to a variable.
$DomainDN=$((Get-ADDomain).DistinguishedName)

# Create the CDP/Teleport container.
# If the command fails with "New-ADObject : An attempt was made to add an object to the directory with a name that is already in use",
# it means the object already exists and you can move on to the next step.
New-ADObject -Name "Teleport" -Type "container" -Path "CN=CDP,CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN"

# Gives Teleport the ability to create LDAP containers in the CDP container.
dsacls "CN=CDP,CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN" /I:T /G "$($SamAccountName):CC;container;"
# Gives Teleport the ability to create and delete cRLDistributionPoint objects in the CDP/Teleport container.

> dsacls "CN=TELEPORT,CN=CDP,CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN" /I:T /G "$($SamAccountName):CCDC;cRLDistributionPoint;"

Reviewer: Not sure if these are case sensitive or not, but the code uses `Teleport`, not `TELEPORT`, so I would make it match just to be safe.

# Gives Teleport the ability to write the certificateRevocationList property in the CDP/Teleport container.
dsacls "CN=TELEPORT,CN=CDP,CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN " /I:T /G "$($SamAccountName):WP;certificateRevocationList;"
# Gives Teleport the ability to create and delete certificationAuthority objects in the NTAuthCertificates container.
dsacls "CN=NTAuthCertificates,CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN" /I:T /G "$($SamAccountName):CCDC;certificationAuthority;"
# Gives Teleport the ability to write the cACertificate property in the NTAuthCertificates container.
dsacls "CN=NTAuthCertificates,CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN" /I:T /G "$($SamAccountName):WP;cACertificate;"

```

## Step 2/6. Prevent the service account from performing interactive logins.

> The Teleport service account is only needed to authenticate over LDAP, meaning that it needn't be able to login to windows machines like an ordinary user.

Reviewer: ```suggestion
The Teleport service account is only needed to authenticate over LDAP, meaning that it needn't be able to login to Windows machines like an ordinary user.
```

Restrict it from doing so by creating a new Group Policy Object (GPO) linked to your entire domain, and then deny it interactive login.

### Create a GPO

1. Open the `Start` menu and run `Group Policy Management`.
2. On the left pane, navigate to `$FOREST > Domains > $DOMAIN`, selecting your forest and domain names respectively.
3. Right click on `$DOMAIN` and select `Create a GPO in this domain, and Link it here...` from the context menu.

> 4. Give your GPO a name like `Block Interactive Login` and ensure `(none)` is selected for `Source Starter GPO`, then hit `OK`.

Reviewer: `Block Interactive Login` might be too vague, as there's nothing that indicates this is specific to the Teleport service account.

Author: How about `Block teleport-svc Interactive Login`

Reviewer: Works for me!

<Figure align="left" bordered caption="Create and Link GPO">
  ![Create and Link GPO](../../img/desktop-access/create-and-link-gpo.png)
</Figure>

### Deny interactive login

1. From the `Group Policy Management` window, right-click on the GPO you just created (`Block Interactive Login`) and select `Edit...` from the context menu.
2. Select:

```text
Computer Configuration > Policies > Windows Settings > Security Settings > Local Policies > User Rights Assignment
```

3. Double click `Deny log on locally` and in the popup, check `Define these policy settings`.
4. Then click `Add User or Group...`, `Browse ...`, enter the SAM account name of the user you created above (`svc-teleport`) and hit `Check Names` select your Group, and then hit `OK` on all the windows.
5. Repeat steps 3 and 4 for `Deny log on through Remote Desktop Services` (in lieu of `Deny log on locally`).

<Figure align="left" bordered caption="Deny Interactive Login">
  ![Deny Interactive Login](../../img/desktop-access/deny-interactive-login.png)
</Figure>

## Step 3/6: Configure a GPO to allow Teleport connections.

> Next, we need to configure a GPO to tell your computers to trust Teleport's CA, allow the certificate-based mechanism that Teleport uses under the hood

Reviewer: ```suggestion
Next, we need to configure a GPO to allow Teleport desktop sessions. This includes telling your computers to trust Teleport's CA, allowing the certificate-based smart card authentication, and ensuring RDP is enabled.
```

(smart cards), and allow Teleport to make RDP connections.

### Export The Teleport CA

<Admonition type="note" title="Requires Existing Cluster">
  The following step requires an existing cluster. If you don't already have a
  Teleport cluster up and running, see our general [Getting
  Started](https://goteleport.com/docs/getting-started/) guide.
</Admonition>

> 1. From the machine running your existing Teleport cluster, get the Teleport user CA certificate by running:

Reviewer: `tctl` doesn't necessarily need to run on the same instance as `teleport` anymore. I would leave this as-is.

```
$ tctl auth export --type=windows > user-ca.cer
```

> 2. Transfer the `user-ca.cer` file to your Domain Controller.

Reviewer: Group policy can be administered from any Windows host with remote server administration tools (RSAT) installed. It does not necessarily need to be a domain controller. In fact, it might even be best practice to _not_ log on to domain controllers directly.

```suggestion
2. Transfer the `user-ca.cer` file to the Windows machine where you can manage group policy.
```

### Create Another GPO and Import the Teleport CA

<Admonition type="note" title="Domain Wide Policy">

>   For the purposes of this example, we apply the GPO we are about to create to

Reviewer: ```suggestion
  For the purposes of this guide, we apply the GPO we are about to create to
```

  our entire AD domain. In the case where you wish for only a subset of
  computers within your AD domain to be accessible via Teleport, you should
  apply the GPO to an OU that includes only such computers.
</Admonition>

> 1. Back on your Domain Controller, create a new GPO like you did in step 2 above, this time giving it a name like `Teleport Access Policy`.

Reviewer: Again, I wouldn't mention domain controller.

# Gives Teleport the ability to create and delete cRLDistributionPoint objects in the CDP/Teleport container.
dsacls "CN=Teleport,CN=CDP,CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN" /I:T /G "$($SamAccountName):CCDC;cRLDistributionPoint;"
# Gives Teleport the ability to write the certificateRevocationList property in the CDP/Teleport container.
dsacls "CN=Teleport,CN=CDP,CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN " /I:T /G "$($SamAccountName):WP;certificateRevocationList;"
# Gives Teleport the ability to create and delete certificationAuthority objects in the NTAuthCertificates container.
dsacls "CN=NTAuthCertificates,CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN" /I:T /G "$($SamAccountName):CCDC;certificationAuthority;"
# Gives Teleport the ability to write the cACertificate property in the NTAuthCertificates container.
dsacls "CN=NTAuthCertificates,CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN" /I:T /G "$($SamAccountName):WP;cACertificate;"

```

## Step 2/6. Prevent the service account from performing interactive logins

<Admonition type="note" title="gpupdate.exe">

>   Throughout this step and the next one, you will be modifying GPO's, and

Reviewer: ```suggestion
  Throughout this step and the next one, you will be modifying GPOs, and
```

>   sometimes GPO modifications can take some time to take effect. You can force

Reviewer: ```suggestion
  sometimes GPO modifications can take some time to propagate to all hosts. You can force
```

>   your GPO changes to take effect immediately at any time by opening a

Reviewer: Note that this only forces the GPO changes to synchronize on the current host, not across the entire fleet. Is there a better way to word this?

```text
Computer Configuration > Policies > Windows Settings > Security Settings > Local Policies > User Rights Assignment
```

3. Double click `Deny log on locally` and in the popup, check `Define these policy settings`.
4. Then click `Add User or Group...`, `Browse ...`, enter the SAM account name of the user you created above (`svc-teleport`) and hit `Check Names` select your Group, and then hit `OK` on all the windows.
5. Repeat steps 3 and 4 for `Deny log on through Remote Desktop Services` (in lieu of `Deny log on locally`).

<Figure align="left" bordered caption="Deny Interactive Login">
  ![Deny Interactive Login](../../img/desktop-access/deny-interactive-login.png)
</Figure>

## Step 3/6: Configure a GPO to allow Teleport connections

Next, we need to configure a GPO to allow Teleport desktop sessions. This includes telling your computers to trust Teleport's CA, allowing the certificate-based smart card authentication, and ensuring RDP is enabled.

### Export the Teleport CA

<Admonition type="note" title="Requires Existing Cluster">
  The following step requires an existing cluster. If you don't already have a
  Teleport cluster up and running, see our general [Getting

>   Started](https://goteleport.com/docs/getting-started/) guide.

Reviewer 2: Please use local file notation instead of global URL notation when referring another article `[Example](../path/to/file.mdx)`

```

2. Right click on `Inbound Rules` and select `New Rule...`.
3. Under `Predefined` select `Remote Desktop`.
4. Only select the rule for `User Mode (TCP-in)`.
5. On the next screen, select `Allow the connection` and finish.

<Figure align="left" bordered caption="Open the Firewall">
  ![Open the Firewall](../../img/desktop-access/firewall.png)
</Figure>

### Allow remote RDP connections

1. Next, select:

```text
Computer Configuration > Policies > Administrative Templates > Windows Components > Remote Desktop Services > Remote Desktop Session Host > Connections
```

2. Right click on `Allow users to connect remotely by using Remote Desktop Services` and select `Edit`. Select `Enabled` and `OK`.

3. Select:

```text
Computer Configuration > Policies > Administrative Templates > Windows Components > Remote Desktop Services > Remote Desktop Session Host > Security
```

4. Right click `Require user authentication for remote connections by using Network Level Authentication`, edit, select **`Disable`** and `OK`.

<Figure align="left" bordered caption="Disable Require user authentication...">
  ![Disable Require](../../img/desktop-access/disable.png)
</Figure>

> ## Step 4/6. Export your Domain Controller's CA certificate.

Reviewer: ```suggestion
## Step 4/6. Export your LDAP CA certificate.
```

## Step 4/6. Export your Domain Controller's CA Certificate

Teleport connects to your Domain Controller via LDAPS. This means that you must let Teleport know that the certificate sent
by your Domain Controller during the initial SSL connection is trusted. If your Domain Controller's certificate is trusted by

> the system repository on the system running Teleport, you can skip this step. Atlernately, if you trust your network, you can

Reviewer: We should make a few small tweaks here:

1. We never actually mention the config field that contains the LDAPS CA certificate.
2. Typo in `Alternately` - we should say `Alternatively`
3. I wouldn't say "if you trust your network" for the `insecure_skip_verify` part. We don't want to encourage the use of this setting for anything other than initial setup or POCs. Maybe say something like "If you are unable to acquire the LDAP CA certificate, you can skip TLS verification by setting `insecure_skip_verify: true`. We do not recommend skipping TLS verification in production environments."

    windows_desktop_logins: ["Administrator"]
```

> See the [RBAC section](https://goteleport.com/docs/ver/8.0/desktop-access/reference/#rbac) in the Reference documentation

Reviewer: Do we usually use a versioned link when linking to other parts of the docs? What happens when 9.0 comes out, do we update the link to point to `/ver/9.0/desktop-access`?

@klizhentas can probably advise us here.

Reviewer 2: @zmb3 @ibeckermayer instead of using URI, please use "[Description](./path.mdx)".
